collectors/python.d.plugin/alarms/metadata.yaml

```diff
@@ -3,52 +3,141 @@ meta:
   module_name: alarms
   alternative_monitored_instances: []
   monitored_instance:
-    name: python.d alarms
-    link: ''
-    categories: []
-    icon_filename: ''
+    name: Netdata Agent Alarms
+    link: "https://github.com/netdata/netdata/blob/master/health/REFERENCE.md#alarm-statuses"
+    categories:
+      - data-collection.notifications
+    icon_filename: ""
   related_resources:
     integrations:
       list: []
   info_provided_to_referring_integrations:
-    description: ''
-  keywords: []
+    description: "Monitor Netdata Agent alarm statuses."
+  keywords:
+    - notification monitoring
+    - alerts
+    - agent alerts
   most_popular: false
 overview:
   data_collection:
-    metrics_description: ''
-    method_description: ''
+    metrics_description: |
+      This collector monitors the statuses of the alarms the Netdata Agent produces and will display them as integer values in a chart.
+    method_description: |
+      It uses the `/api/v1/alarms?all` endpoint of the Agent, and creates a dimension per alarm.
   supported_platforms:
     include: []
     exclude: []
   multi-instance: true
   additional_permissions:
-    description: ''
+    description: ""
   default_behavior:
     auto_detection:
-      description: ''
+      description: |
+        It discovers instances of Netdata running on localhost, and gathers metrics from `http://127.0.0.1:19999/api/v1/alarms?all`. `CLEAR` status is mapped to `0`, `WARNING` to `1` and `CRITICAL` to `2`. Also, by default all alarms produced will be monitored.
```

Comment:
We definitely need to rework this collector, mapping to integers is broken when aggregating over time. We probably shouldn't even suggest using it until it's fixed.

Reply:
probably cc @andrewm4894

Reply:
(also, let's also mention the naming scheme, alarms -> alerts.)

Reply:
hot take: this collector is actually the best visualisation of alerts in NC right now: 

![image](https://github.com/netdata/netdata/assets/2178292/4f8ca700-29a2-410e-95b0-21f81ce463d2)

agging over status integers is fine and actually good as lets you easily see diff of a status=2 that was short vs one that was long in time

Reply:
It is not fine, it is broken for the reason I listed above. Do you want me to elaborate?

Reply:
yes - is it not just aggregating the integers over time eg [0,1,1,1,2,2,1,1,0] 

this is fine: 

![image](https://github.com/netdata/netdata/assets/2178292/da3673de-e216-46d8-b8d2-b17465583b45)

Or is there some deeper technical reason its not actually able to just average the intgers?

Reply:
It is broken because

- any aggregation (over time) method (avg, mix, max) produces meaningless and misleading values.
- any aggregation (over nodes) method produces meaningless and misleading values.

In case you are wondering what I would call a proper way - the same we do for any other state/status metric:

 - a chart per instance (in your case - per alarm)
 - a dimension per state (critical, warning, etc). 1 - active state, 0 - inactive state.

Having that will give nice aggregation over nodes (the default will be the number of alarms in each state) and meaningful over time (avg will be % of time spent in the state in the selected time range, min/max - was or was not this alarm in the state in the selected time range).

Reply:
i think flip side of that then would be needing a chart per status and so not be able to see both status on same chart to easily see transition from warning to critical etc. 

although maybe now with new cloud charts and instances like you say could be a way. 

for me being able to see one metric that encodes the status and covers both warning and critical state on one metric is very useful as opposed to it ending up as two different lines with two different colors on the chart.

i think i get what you are saying though. 

is there any other collectors that are similar to like you suggest - eg have a status metrics over lots of dims?

Reply:
It is not what I am saying/my opinion, it is a fact and a problem - when you create charts like in python.d/alarms the values are just wrong (over time, over nodes). It may be useful for you personally, but we shouldn't have such charts in core (or have them but consider them as a bug and fix).


> eg have a status metrics over lots of dims?

A lot, e.g. systemdunits, network interface operstate/carrier, ipmi sensors.

Reply:
> active status value is 1, and the rest have 0

"active status" would be either "warning" or "critical" ?

Reply:
>  active status value is 1

so this would map "warning" and "critical" to 1?